### PR TITLE
ci: add github workflow

### PR DESCRIPTION
**Pull request by RaphaelPour**

### CODE REVIEW

File: .github/workflows/ruby-rspec.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: hcloud-ruby ci
+on: [push]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: [ '2.7' ]
+
```
Comment (Kjarrigan) [inline .github/workflows/ruby-rspec.yml:9]: I would add 3.1 too and set it to continue-on-error
```yaml
strategy:
  matrix:
    ruby-version: ['2.7', '3.1']
continue-on-error: ${{ matrix.ruby == '3.1' }}
```

File: .github/workflows/ruby-rspec.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+      - name: Run double tests
+        run: bundle exec rspec -t doubles --order rand
+      - name: Run legacy tests
+        run: LEGACY_TESTS=y bundle exec rspec -t ~doubles
+      - name: Run rubocop
```
Comment (Kjarrigan) [inline .github/workflows/ruby-rspec.yml:21]: I would move this to a second job. I don't fancy mixing lint and spec. Either use the predefined Rubocop-Action or just do something like:
```yaml
jobs:
  lint:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        ruby-version: [ '2.7' ]
   steps:
      - uses: actions/checkout@v2
      - name: Setup Ruby
        uses: ruby/setup-ruby@v1
        with:
          ruby-version: ${{ matrix.ruby-version }}
          bundler-cache: true
      - name: Run rubocop 
        run: bundle exec rubocop
```

You could even use a slimmer base-image than ubuntu and probably also skip the bundler part. Just install and execute rubocop withouth anything else (like we do it in some other projects)

Comment (RaphaelPour) [reply .github/workflows/ruby-rspec.yml]: I've also added 3.0.

File: .github/workflows/ruby-rspec.yml (hunk 1 above)
Comment (RaphaelPour) [reply .github/workflows/ruby-rspec.yml:20]: I've also added 3.0.

**Review by Kjarrigan — APPROVED** [review]
LGTM

**Pull request merged**